lib/jxl/enc_file.cc

```diff
@@ -139,6 +136,7 @@ Status WriteHeaders(CodecMetadata* metadata, BitWriter* writer,
   return true;
 }
 
+// TODO(firsching): remove this function and all tests that use it.
```

Comment:
"all tests that use it" is basically all roundtrip tests, I don't think we want to remove those

The plan should be to re-implement this using API functions and move it to test_utils.h, same as with DecodeFile